pkg/_pub_shared/lib/data/package_api.dart

```diff
@@ -85,17 +85,25 @@ class GithubPublishingConfig {
   String? tagPattern;
 
   /// Whether to require the action from a specific environment.
-  bool? requireEnvironment;
+  late bool requireEnvironment;
 
   /// The GitHub environment where the publishing is required from.
   String? environment;
 
+  /// Whether the (default) `push` event is enabled.
+  late bool isPushEventEnabled;
+
+  /// Whether the `workflow_dispatch` event is enabled.
+  late bool isWorkflowDispatchEventEnabled;
+
   GithubPublishingConfig({
-    this.isEnabled,
+    this.isEnabled = false,
```

Comment:
Have you considered using `@JsonKey(disallowNullValue: true, defaultValue: false)` annotation instead?

Reply:
`disallowNullValue` would enforce us to do a migration. In its current form we may just use it as-is, and we may or may not do a migration, it is just fine to let the migration happen for users that change their settings in any way.

Reply:
Ah, yeah, that's fine. Let's not use `disallowNullValue` :rofl: 


-------------

I do think it'd be wise to do a migration eventually, it easily becomes much harder to not make mistakes if we don't.